src/components/App.tsx

```diff
@@ -502,6 +503,16 @@ class App extends React.Component<any, AppState> {
       this.initializeSocketClient({ showLoadingState: true });
     }
 
+    if (
+      this.state.editingLinearElement &&
+      !this.state.selectedElementIds[this.state.editingLinearElement.elementId]
+    ) {
+      // defer so that the commitToHistory flag isn't reset via current update
```

Comment:
ewwwwwwwww, we really need to fix how undo history works :)

Reply:
although this hack isn't related to history per se